src/main/java/de/azapps/kafkabackup/common/partition/cloud/S3PartitionWriter.java

```diff
@@ -54,6 +54,9 @@ private void writeToBatch(Record record) {
       if (batchWriter == null) {
         batchWriter = new S3BatchWriter(awsS3Service, bucketName, topicPartition, record);
       } else {
+        if (record.kafkaOffset() <= batchWriter.getEndOffset()) {
+          log.debug("Skipping message already added to batch.");
```

Comment:
Shouldn't we return here then, and skip lines 51-53? 🤔

Reply:
Ofcourse :-)